lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -992,18 +994,26 @@ void removeNonExistingTasksFromTaskIdList(
       LOGGER.debug(
           "entry to augmentTaskSummariesByContainedSummaries(taskSummaries= {})", taskSummaries);
     }
+    // splitting Augmentation into steps of maximal 32000 tasks
+    // reason: DB2 has a maximum for parameters in a query
+    List<TaskSummary> result =
+        ListUtil.partitionBasedOnSize(taskSummaries, 32000).stream()
+            .map(this::augmentTaskSublist)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
 
-    List<TaskSummary> result = new ArrayList<>();
-    if (taskSummaries == null || taskSummaries.isEmpty()) {
-      return result;
-    }
+    LOGGER.debug("exit from to augmentTaskSummariesByContainedSummaries()");
+    return result;
+  }
 
+  private List<TaskSummaryImpl> augmentTaskSublist(List<TaskSummaryImpl> taskSummaries) {
```

Comment:
I don't like this name. Maybe we can call this `augmentTaskSummariesByContainedSummariesWithoutPartitioning` and the parent method `augmentTaskSummariesByContainedSummariesWithPartitioning`

Reply:
What about "augmentTaskSummariesByContainedSummaries" and "augmentTaskSummariesByContainedSummariesForSublist"? Or do you prefer your suggestions? I'm not a huge fan of complex method names...